src/chrome/chromeConnection.ts

```diff
@@ -128,9 +129,19 @@ export class ChromeConnection implements IObservableEvents<IStepStartedEventsEmi
     /**
      * Attach the websocket to the first available tab in the chrome instance with the given remote debugging port number.
      */
-    public attach(address = '127.0.0.1', port = 9222, targetUrl?: string, timeout?: number, extraCRDPChannelPort?: number): Promise<void> {
-        return this._attach(address, port, targetUrl, timeout, extraCRDPChannelPort)
-            .then(() => { });
+    public attach(address = '127.0.0.1', port = 9222, targetUrl?: string, timeout = ChromeConnection.ATTACH_TIMEOUT, extraCRDPChannelPort?: number, userDataDir?: string): Promise<void> {
```

Comment:
Is there any way to only implement this functionality for Chrome?
This doesn't make sense for Node.js

Reply:
True, but it's already optional, right? Will node ever send --user-data-dir?

Reply:
resolved with refactoring anyway